### PR TITLE
#96 - Modernization - Phase 1 - ES6 Modules - 3

### PR DESCRIPTION
**Pull request by parthopdas**
#96 - Modernization - Phase 1 - ES6 Modules - 3

> Moving navigation / search / dictionary to module
> Extracting global variables
> Fixing tests
> Cleaning up linter

### CODE REVIEW

File: _dprhtml/features/navigation/init.js [hunk 1]
```diff
@@ -1,206 +1,232 @@
...
+    ViewModel.para(),
+  )
+}
+
+const parseNavigationURLParams = () => {
+  const urlParams = decodeURIComponent(window.location.search).substring(1, window.location.search.length).split('&')
+  let query = ''
+  let para = ''
+  urlParams.forEach((parameter) => {
+    const [psec0, psec1, ..._] = parameter.split('=')
```
Comment (rrogowski) [inline _dprhtml/features/navigation/init.js:199]: If you wanted to eliminate some of the noise, I believe the following is semantically and functionally equivalent:
```suggestion
    const [psec0, psec1] = parameter.split('=')
```

Comment (rrogowski) [inline _dprhtml/features/navigation/init.js:199]: If you wanted to eliminate a little bit of ​noise, I believe the following is semantically equivalent:
```suggestion
    ​const [psec0, psec1] = parameter.split('=')
``

File: _dprhtml/features/search/init.js [hunk 2]
```diff
@@ -1,154 +1,168 @@
...
+    if (this.searchBookString() === '') {
+      return true
     }
+
+    const bookArray = (this.searchBookString() || '').split(',')
+    return bookArray.includes(`${bookNumber}`)
   }
 
-  searchSet(set){
-    set = set.toLowerCase();
```
Comment (rrogowski) [inline _dprhtml/features/search/init.js:79]: I was just talking with someone the other day about why re-assigning parameters can be confusing. Thanks for cleaning this up!

File: _dprhtml/features/other-dialogs/init.js [hunk 3]
```diff
@@ -1,163 +1,170 @@
...
+    newNodeScroll.appendChild(tScroll);
+    newNode.appendChild(newNodeScroll);
+
+    var tDesc = xmlDoc.createTextNode(desc);
+    newNodeDesc.appendChild(tDesc);
+    newNode.appendChild(newNodeDesc);
+
+    xmlDoc.documentElement.appendChild(newNode);
+
+    //Navigation.ViewModel.updateBookmarks(xmlDoc);
```
Comment (rrogowski) [inline _dprhtml/features/other-dialogs/init.js:140]: Do we want to leave this comment in?

**Review by rrogowski — APPROVED** [review]

File: _dprhtml/features/navigation/init.js (hunk 1 above)
Comment (parthopdas) [reply _dprhtml/features/navigation/init.js]: fixed

File: _dprhtml/features/other-dialogs/init.js (hunk 3 above)
Comment (parthopdas) [reply _dprhtml/features/other-dialogs/init.js]: removed.

**Review by catalin-i — APPROVED** [review]
Looks great!

**Pull request merged**